.github/workflows/govulncheck.yml

```diff
@@ -0,0 +1,100 @@
+# This workflow is meant to run govulncheck on all the branches
+# that are containing a maintained version of OpenTofu.
+# For more considerations about this, check this PR: https://github.com/opentofu/opentofu/pull/2600
+#
+# This will attempt to create an issue with the results for each scanned version
+# in case any vulnerability is found. If the issue for the scanned branch already exists
+# the run will just print the scan output and will leave the issue untouched.
+#
+# This is meant to run _only_ from the main branch, on a scheduled manner.
+# All the other branches will be scanned directly by the run triggered from the main branch.
+
+name: Govulncheck
+
+on:
+  schedule:
+    - cron: '42 3 * * SUN'
+  workflow_dispatch: {}
+
+jobs:
+  gather-branches:
+    name: Collect the branches that needs to be scanned
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    env:
+      GH_TOKEN: ${{ github.token }}
+    outputs:
+      branches: ${{ steps.branches.outputs.branches }}
+    steps:
+      - id: branches
+        run: |
+          ## This is using GitHub API and we manually filter out the versions that are not supported anymore
+          # gh api "repos/opentofu/opentofu/branches?protected=true" --paginate | tee out 
+          # cat out | jq '.[] | select(.name != "v1.6") | [.name]' | jq -sc 'add' | tee branches
+          # echo "branches=$(cat branches)">> "$GITHUB_OUTPUT"
+          
+          ## This is using https://endoflife.date since there 1.6 is already marked as it reached EOL
+          ## This is adding manually the main branch to scan it as well
+          curl https://endoflife.date/api/opentofu.json | tee out 
+          cat out | jq '.[] | select(.eol == false) | [.cycle]' | jq -sc 'add' | jq -c '["main"] + .' | tee branches
+          echo "branches=$(cat branches)">> "$GITHUB_OUTPUT"
+  govulncheck:
+    name: Run govulncheck for ${{ matrix.branch }}
+    needs: gather-branches
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    env:
+      GH_TOKEN: ${{ github.token }}
+    strategy:
+      matrix:
+        branch: ${{ fromJSON(needs.gather-branches.outputs.branches )}}
+      fail-fast: false
+# Or we can configure this manually here, since this workflow will be executed on a schedule only from the main branch
+#    strategy:
+#      matrix:
+#        include:
+#          - { branch: main }
+#          - { branch: v1.9 }
+#          - { branch: v1.8 }
+#          - { branch: v1.7 }
+#      fail-fast: false
```

Comment:
This is the 3rd approach on configuring what versions of OpenTofu we want to scan.
Pros:
* Clearly configured and easy to maintain.

Cons:
* Manual maintenance, but still nicer than the first approach.

Reply:
I think this is my preference, we can easily add that to the release process notes.  The third party option is nice, but I prefer to have control over the filter in this repo.